record_generator.go

```diff
@@ -26,15 +26,33 @@ import (
 	"github.com/google/uuid"
 )
 
-type recordGenerator interface {
-	generate() (sdk.Record, error)
+type recordGenerator struct {
+	config RecordConfig
+	cached sdk.RawData
 }
 
-type defaultRecordGen struct {
-	config RecordConfig
+func newRecordGenerator(config RecordConfig) recordGenerator {
+	return recordGenerator{config: config}
+}
+
+func (g *recordGenerator) warmUp() error {
```

Comment:
Nitpick: functions such as this one are usually called "init" in Go.

Reply:
:+1:  Done.